### ISSUE TITLE
Spot removal tool: Spot button toggles profiled lens correction on/off when tool is disabled

### ISSUE THREAD

**Issue by waynesutton50** [Beep6581/RawTherapee#6242]
The spot-removal button toggles the profiled lens correction on/off, even if the spot-removal tool itself is turned off. 

Steps to reproduce:

-Load an image that has visible lens-distortion correction when the profiled lens correction is turned on
-Go to the Spot removal tool in the Detail tab and toggle the spot button. The tool itself does not have to be enabled.

Tested with AppImage RawTherapee_dev_5.8-2926-gd11d834_20210517.AppImage and with a locally compiled version 
5.8-2927-gbd9704da8

EDIT: The same behaviour can be observed in ART 1.7.1 but this seems to have been corrected in 1.8.4

**Comment by heckflosse** [comment]
Confirmed

**Comment by ghost** [comment]
Is this a bug?

I've noticed this behaviour before and I always believed that disabling lens correction was needed because spot removal comes before lens correction in the pipeline and to  be able to do its job correctly lens correction needs to be disabled for the moment. This is an assumption of mine, though, spot removal isn't yet mentioned on the RawPedia Toolchain Pipeline page.

The spot removal pop-up does mention this though (last paragraph):

> [...] right click outside any spot to end the spot editing mode

This turns off editing mode and restores the lens correction.

I think that turning it off is part of the process and leaving it on is user error. No?

EDIT: I just read this part:

> The same behaviour can be observed in ART 1.7.1 but this seems to have been corrected in 1.8.4

Which seems to prove me wrong.

**Comment by heckflosse** [comment]
> Is this a bug?
I've noticed this behaviour before and I always believed that disabling lens correction was needed because spot removal comes before lens correction in the pipeline and to be able to do its job correctly lens correction needs to be disabled for the moment.

The bug is, that lens correction is disabled even if spot removal is disabled while you toggle the spot button

**Comment by ghost** [comment]
So my assumption is correct but I'm way of the mark here nonetheless... I'll shut up now :zipper_mouth_face:

**Comment by waynesutton50** [comment]
I've just checked ART 1.8.4 again. The spot button doesn't disable the lens correction no matter what state the tool is in (enabled or disabled)

**Comment by AndyAstbury** [comment]
Whether my problem with the spot tool is connected here or not I don't know but I did a small [vid showing my problem ](https://youtu.be/aZ-HTog-2UA)

Or is just me displaying operator error?

**Comment by ghost** [comment]
@AndyAstbury : This is not related to the lens correction issue it seems.

I'm not able to replicate your issue of placing the source spot on Linux (Buster/Debian 10.9) with the very latest RawTherapee (version 5.8-2959-g7840cdea8).

As far as I can tell it isn't user error: Set destination circle, grab circle holding left mouse button and move it. Let go of left mouse button. Done. Then you could relocate either circle in the same way (left click on it, hold, move and let go).

I have seen *ghost trails* in the past but have not seen them lately.

You are running Linux in a VM on a Mac? Maybe someone with the same setup can confirm this one?

**Comment by Lawrence37** [comment]
This behavior is intentional as far as I know. Like every other tool, it is possible to make adjustments even if the tool is disabled. You can edit the spots with the spot removal disabled. Since the spot removal is applied before lens and geometry, they are disabled to give an accurate way to position the spots. Whether the behavior with the tool disabled is user-friendly or not, I do not know.

**Comment by Lawrence37** [comment]
@AndyAstbury I fixed a visual bug for the linking line. I will look at the other problems later.

Edit: I did not notice the other issues on my build, so this might take some time to fix. Could you please try adding spots with Ctrl + click-and-drag instead of a simple ctrl + click and see if that also causes issues?

**Comment by AndyAstbury** [comment]
@Lawrence37 Just updated to latest build, tried it with my usual Bluetooth Kensington Trackball and have exactly the same problem.
However, if I use my wired Wacom it works perfectly and I can move both spots and they remain joined by the linking line.
Just tried using my Mac Bluetooth mouse and have the same problems, so it appears to be a un-wired mouse problem with either RT, Linux or my Parallels though both bluetooth devices function fully in GIMP.

**Comment by Thanatomanic** [comment]
> This behavior is intentional as far as I know. Like every other tool, it is possible to make adjustments even if the tool is disabled.

I agree that this is not a bug, but intentional behavior. The same thing happens with e.g. the Graduated Filter and Local adjustments: even if those tools are off, you can still modify their on-preview widgets. The only difference with the spot removal tool is that it _has_ to work before lens corrections and will automatically disable them.
Changing this behavior for one tool only makes little sense imo, it introduces inconsistencies in the GUI. 

Edit: we might need to add info to the tooltip explaining that lens corrections will be temporarily disabled during spot editing.

**Comment by Thanatomanic** [comment]
Also,
> EDIT: The same behaviour can be observed in ART 1.7.1 but this seems to have been corrected in 1.8.4

I've just tested `ART 1.8.4-25-g1db84c6f4` and toggling the spot editor also toggles the automatic lens correction. The only difference ART has, is that when you disable the tool, the spot editor is closed as well and the lens corrections are enabled again. In RawTherapee the spot editor remains active.
I am not in favor of copying the behavior from ART, unless we also make it so that when you enable the spot editor you also enable the entire tool. But this is also different from the rest of the GUI (with the possible exception of the White Balance tool).

**Comment by waynesutton50** [comment]
> I've just tested `ART 1.8.4-25-g1db84c6f4`

I tested an earlier 1.8.4 commit (b0bdd1fda675) so maybe there have been changes since then. However, the real concern is that switching 'off' the Spot Removal power button doesn't restore the profiled lens correction if the spot button has been toggled on and no further action taken. This isn't a problem if the lens distortion is obvious (eg significant barrel distortion) but this isn't always the case.  It's also worth noting that clicking on the LA tab restores the profiled lens correction.

**Comment by waynesutton50** [comment]
> The only difference ART has, is that when you disable the tool, the spot editor is closed as well and the lens corrections are enabled again

I have recompiled the latest version and can confirm this. 



> The same thing happens with e.g. the Graduated Filter and Local adjustments: even if those tools are off, you can still modify their on-preview widgets.

Sorry I probably didn't explain the problem very well. I am referring to the situation where you press the spot removal button but you don't actually create a widget. If you do this and forget to press the button again you disable the lens correction no matter what state the power button is in and if the distortion isn't that obvious, you may not notice.
